tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py

```diff
@@ -0,0 +1,421 @@
+import logging
+import uuid
+
+import boto3
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_aws_creds_are_missing,
+    tier2,
+)
+from ocs_ci.framework.testlib import E2ETest, skipif_ocs_version
+from ocs_ci.ocs import bucket_utils
+from ocs_ci.ocs import constants
+
+logger = logging.getLogger(__name__)
+
+OBJ_DATA = "Sample string content to write to a S3 object"
+
+
+def setup_objects_to_list(mcg_obj, bucket_name, amount=100, prefix=""):
+    """
+    Prepares two directories and populate one of them with objects
+
+     Args:
+        mcg_obj (obj): MCG object
+        amount (int): Number of test objects to create
+        bucket_name (str): Name of the bucket
+        prefix (str): Name of the prefix
+    Returns:
+        Tuple: Returns tuple containing the keys, prefixes and index
+
+    """
+    object_keys = []
+    object_prefixes = []
+    for i in range(amount):
+        obj_key = f"{prefix}-{i}/ObjKey-{i}"
+        bucket_utils.s3_put_object(
+            s3_obj=mcg_obj, bucketname=bucket_name, object_key=obj_key, data=OBJ_DATA
+        )
+        object_keys.append(obj_key)
+        object_prefixes.append(f"{prefix}-{i}/")
+    mid_index = len(object_keys) // 2
+    return object_keys, object_prefixes, mid_index
+
+
+def get_list_and_verify(response, keys, verify="Contents", prefix="", delimiter=""):
```

Comment:
Please add docstring with description what it does.

Reply:
Added doc string

Reply:
Agree. Test is not split into 3 sub tests: Basic ops , object version and multipart.